src/phalanx/models/applications.py

```diff
@@ -0,0 +1,78 @@
+"""Pydantic models for Phalanx applications."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pydantic import BaseModel
+
+from .secrets import RequiredSecret, Secret
+
+__all__ = [
+    "Application",
+    "ApplicationInstance",
+]
+
+
+class Application(BaseModel):
+    """A Phalanx application."""
+
+    name: str
+    """Name of the application."""
+
+    values: dict[str, Any]
+    """Base Helm chart values."""
+
+    environment_values: dict[str, dict[str, Any]]
+    """Per-environment Helm chart overrides by environment name."""
+
+    secrets: list[Secret]
+    """Base secret configuration for the application."""
+
+    environment_secrets: dict[str, list[Secret]]
+    """Per-environment secrets for the application."""
+
+
+class ApplicationInstance(BaseModel):
+    """A Phalanx application as configured for a specific environment."""
+
+    name: str
+    """Name of the application."""
+
+    environment: str
+    """Name of the environment for which the application is configured."""
+
+    values: dict[str, Any]
+    """Merged Helm values for the application in this environment."""
+
+    secrets: list[RequiredSecret] = []
+    """Secrets required for this application in this environment."""
+
+    def is_condition_met(self, condition: str | None) -> bool:
+        """Determine whether a secret condition has been met.
+
+        Conditions are used both for the secret as a whole and for the
+        ``copy`` and ``generate`` sections. The condition is met if it either
+        is `None` or if it is a string pointing to a values parameter for the
+        application instance that is set to a true value.
+
+        Parameters
+        ----------
+        condition
+            Condition to check.
+
+        Returns
+        -------
+        bool
+            `True` if the condition was met or does not exist, `False`
+            otherwise.
+        """
```

Comment:
I think this method is totally fine, but I wonder if its naming/documentation is almost putting a secrets concern on the application? Like, what if it was named `has_truthy_value(self, value_path: str | None)` and then all the secrets code knows that conditions are based on finding truthy values at a value path?

I'm probably making a mountain out of a molehill, so feel free to ignore this.

Reply:
This is a good idea. I turned this into `is_values_setting_true` and added a wrapper to deal with the null condition case. Thanks!